### ISSUE TITLE
security vulnerability? due to old hawk@3.1.3 (and hoek@2.16.3) dependency

### ISSUE THREAD

**Issue by wvanderdeijl** [mapbox/node-pre-gyp#346]
in our project, snyk recently started complaining that we have a dependency with a known security vulnerability. It complains about `hoek@2.16.3` which is required by `hawk@3.1.3` which is required by the latest version of `node-pre-gyp`

the latest version of `hoek` (version 5.0.3) fixed the vulnerability. But `node-pre-gyp` has locked the version of `hawk` to 3.1.3 while the latest version of `hawk` is 7.0.7. Using such an old version of `hawk` also uses a very old version of `hoek`.

Would it be easy to upgrade to the latest version of `hawk` so we get the latest version of `hoek` without the vulnerability?

More info about the (low prio) vulnerability in `hoek` can be found at https://snyk.io/vuln/npm:hoek:20180212

**Comment by nathany** [comment]
More information on the security vulnerability in old versions of hoek:
https://nodesecurity.io/advisories/566
https://hackerone.com/reports/310439

For what it's worth, the patch was also backported to hoek 4.2.1.
https://github.com/hapijs/hoek/commits/v4.x.x

**Comment by krotscheck** [comment]
This is now also being reported by nsp, https://nodesecurity.io/advisories/566

**Comment by guyo13** [comment]
Or upgrade to hawk@7.0.7 which have the patched version of hoek in the packge.json

**Comment by pumano** [comment]
@springmeyer any news about dependency updates?

**Comment by evheniy** [comment]
Hi
Webpack has dependency 

https://github.com/webpack/webpack/issues/6552

Could you please update hawk to latest versuon.

Thank you!

**Comment by ChALkeR** [comment]
That won't fix things, as there is `+-- request@2.81.0` → `+-- hawk@3.1.3 deduped` dep, which also pulls `hawk@3.1.3` which also pulls in old `hoek`.

Request has several PRs to fix that, but that didn't happen yet.

Perhaps migrate away from `request`?

**Comment by SebastianSchmidt** [comment]
@ChALkeR [See PR: Remove hawk, upgrade request](https://github.com/mapbox/node-pre-gyp/pull/347#issuecomment-366083505)

**Comment by ChALkeR** [comment]
@SebastianSchmidt request itself isn't updated yet, that won't help.

**Comment by ChALkeR** [comment]
@SebastianSchmidt Ah, looks like I was indeed wrong, current request version depends on a patched version — `hoek` backported the fix, I missed that.
```js
     '5.0.0': '2017-09-26T04:22:40.507Z',
     '5.0.1': '2017-10-26T08:22:11.034Z',
     '5.0.2': '2017-11-03T09:12:15.703Z',
     '5.0.3': '2018-02-06T17:55:50.172Z',
     '4.2.1': '2018-02-15T16:47:47.274Z' 
```

**Comment by springmeyer** [comment]
Fixed by merging #347
